src/python/pants/init/engine_initializer.py

```diff
@@ -117,8 +119,13 @@ def run_goal_rules(
 
         workspace = Workspace(self.scheduler_session)
         interactive_runner = InteractiveRunner(self.scheduler_session)
+        # NB: Keep this in sync with the property `goal_param_types`.
+        params = Params(specs, self.console, workspace, interactive_runner)
```

Comment:
Could not see that this had to be created for every loop iteration.. so moved it up front, so I could use it for the dependence analysis too.